src/stepFunctions/commands/visualizeStateMachine/aslVisualizationManager.ts

```diff
@@ -22,7 +22,14 @@ export class AslVisualizationManager {
         return this.managedVisualizations
     }
 
-    public async visualizeStateMachine(globalStorage: vscode.Memento): Promise<vscode.WebviewPanel | undefined> {
+    public async visualizeStateMachine(
+        globalStorage: vscode.Memento,
+        params: {
+            activeTextEditor: vscode.TextEditor | undefined
+        } = {
+            activeTextEditor: vscode.window.activeTextEditor,
+        }
```

Comment:
we could avoid this dance by removing the `| undefined` from the type, and forcing the caller to always specify the argument.

Reply:
Was trying to avoid the inevitable. Will make sure that this doesn't break anything and swap over.